cdap-ranger/cdap-ranger-lookup/src/main/java/co/cask/cdap/security/authorization/ranger/lookup/client/CDAPRangerLookupClient.java

```diff
@@ -142,25 +277,208 @@ private void initConnection() throws IOException {
+    if (datasetModuleClient != null) {
+      for (DatasetModuleMeta datasetSpecificationSummary : datasetModuleClient.list(namespace)) {
+        String name = datasetSpecificationSummary.getName();
+        if (datasetModuleList == null || !datasetModuleList.contains(name)) {
+          datasetModules.add(name);
+        }
+      }
+    } else {
+      LOG.warn("Failed to get Datasets Modules. DatasetModuleClient is not initialized.");
+    }
+
+    if (LOG.isDebugEnabled()) {
+      LOG.debug("<== CDAPRangerLookupClient.getDatasetModules(): " + datasetModules);
+    }
+    return datasetModules;
+  }
+
+  private List<String> getDatasetTypes(@Nullable List<String> datasetTypeList, NamespaceId namespace) throws
+    Exception {
+    if (LOG.isDebugEnabled()) {
+      LOG.debug("==> CDAPRangerLookupClient.getDatasetTypes() ExcludeDatasetModuleList :" + datasetTypeList);
+    }
+
+    List<String> datasetTypes = new ArrayList<>();
+    if (datasetTypeClient != null) {
+      for (DatasetTypeMeta datasetTypeMeta : datasetTypeClient.list(namespace)) {
+        String name = datasetTypeMeta.getName();
+        if (datasetTypeList == null || !datasetTypeList.contains(name)) {
+          datasetTypes.add(name);
+        }
+      }
+    } else {
+      LOG.warn("Failed to get Datasets Types. DatasetTypeClient is not initialized.");
+    }
+
+    if (LOG.isDebugEnabled()) {
+      LOG.debug("<== CDAPRangerLookupClient.getDatasetTypes(): " + datasetTypes);
+    }
+    return datasetTypes;
+  }
+
+  private List<String> getSecureKeys(@Nullable List<String> secureKeyList, NamespaceId namespace) throws Exception {
+    if (LOG.isDebugEnabled()) {
+      LOG.debug("==> CDAPRangerLookupClient.getSecureKeys() ExcludeArtifactList :" + secureKeyList);
+    }
+
+    List<String> secureKeys = new ArrayList<>();
+    if (secureStoreClient != null) {
+      for (String secureKey : secureStoreClient.listKeys(namespace).keySet()) {
+        if (secureKeyList == null || !secureKeys.contains(secureKey)) {
+          secureKeys.add(secureKey);
+        }
+      }
+    } else {
+      LOG.warn("Failed to get Secure Keys. SecureStoreClient is not initialized.");
+    }
+
+    if (LOG.isDebugEnabled()) {
+      LOG.debug("<== CDAPRangerLookupClient.getSecureKeys(): " + secureKeys);
+    }
+    return secureKeys;
+  }
+
+  private List<String> getArtifacts(@Nullable List<String> artifactList, NamespaceId namespace) throws Exception {
+    if (LOG.isDebugEnabled()) {
+      LOG.debug("==> CDAPRangerLookupClient.getArtifacts() ExcludeArtifactList :" + artifactList);
+    }
+
+    List<String> artifacts = new ArrayList<>();
+    if (artifactClient != null) {
+      for (ArtifactSummary artifactSummary : artifactClient.list(namespace)) {
+        String name = artifactSummary.getName();
+        String version = artifactSummary.getVersion();
+        if (artifactList == null || !artifacts.contains(name)) {
+          artifacts.add(name + "(Version: " + version + ")");
```

Comment:
We decided we will have version when the code was written. The requirement changed now. I